### ISSUE TITLE
Question: how to use GroupBy in specification?

### ISSUE THREAD

**Issue by chkjamil** [dotnet-architecture/eShopOnWeb#240]
Hi,
I wanted to use group data at database level by using Specification, so how could I use GroupBy clause (in Specification) to achieve this?
Regards,

**Comment by efleming18** [comment]
Hey @chkjamil - I created [this PR](https://github.com/dotnet-architecture/eShopOnWeb/pull/243) to see if this is what you're after? In this PR, I added a GroupBy specification to the catalog filter so it would be easy to see if it was working or not. After adding this, the first page of the catalog looks like this. 
![image](https://user-images.githubusercontent.com/1899987/56869928-acd37300-69d5-11e9-9594-de4d07b4d3b5.png)

If you notice the above image has the items grouped by `CatalogType`. This one groups by shirts, mugs, then sheets. 

Does this help at all?
